From 3d5b0fdc57883a38ed314033f0b4f121584926d5 Mon Sep 17 00:00:00 2001
From: Kevin Zhang <k.zhang@freescale.com>
Date: Sun, 16 Mar 2008 20:18:45 -0600
Subject: [PATCH] ENGR00068855 Add missing .valid function for power management

Add missing .valid function for power management due to kernel upgrade.
It was fixed in the 2.6.22 but was dropped when upgrading to 2.6.24.
This fix impacts on MX27, MX31 and MXC91321 based platforms.

Signed-off-by: Kevin Zhang <k.zhang@freescale.com>
---
 arch/arm/mach-mx27/pm.c     |    8 +++++++-
 arch/arm/mach-mx3/pm.c      |    8 +++++++-
 arch/arm/mach-mxc91321/pm.c |    9 +++++++++
 3 files changed, 23 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-mx27/pm.c b/arch/arm/mach-mx27/pm.c
index 7fc91b7..6e1e941 100644
--- a/arch/arm/mach-mx27/pm.c
+++ b/arch/arm/mach-mx27/pm.c
@@ -15,7 +15,7 @@
  * Cleanup 2004 for OMAP1510/1610 by Dirk Behme <dirk.behme@de.bosch.com>
  *
  * Modified for the MX27
- * Copyright 2007 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2007-2008 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * This program is free software; you can redistribute it and/or modify it
  * under the terms of the GNU General Public License as published by the
@@ -79,7 +79,13 @@ static void mx27_suspend_finish(void)
 	return;
 }
 
+static int mx27_pm_valid(suspend_state_t state)
+{
+	return (state > PM_SUSPEND_ON && state <= PM_SUSPEND_MAX);
+}
+
 struct platform_suspend_ops mx27_suspend_ops = {
+	.valid = mx27_pm_valid,
 	.prepare = mx27_suspend_prepare,
 	.enter = mx27_suspend_enter,
 	.finish = mx27_suspend_finish,
diff --git a/arch/arm/mach-mx3/pm.c b/arch/arm/mach-mx3/pm.c
index 27f78a2..a2c672f 100644
--- a/arch/arm/mach-mx3/pm.c
+++ b/arch/arm/mach-mx3/pm.c
@@ -15,7 +15,7 @@
  * Cleanup 2004 for OMAP1510/1610 by Dirk Behme <dirk.behme@de.bosch.com>
  *
  * Modified for the MX31
- * Copyright 2005-2007 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2005-2008 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * This program is free software; you can redistribute it and/or modify it
  * under the terms of the GNU General Public License as published by the
@@ -79,7 +79,13 @@ static void mx31_suspend_finish(void)
 	return;
 }
 
+static int mx31_pm_valid(suspend_state_t state)
+{
+	return (state > PM_SUSPEND_ON && state <= PM_SUSPEND_MAX);
+}
+
 struct platform_suspend_ops mx31_suspend_ops = {
+	.valid = mx31_pm_valid,
 	.prepare = mx31_suspend_prepare,
 	.enter = mx31_suspend_enter,
 	.finish = mx31_suspend_finish,
diff --git a/arch/arm/mach-mxc91321/pm.c b/arch/arm/mach-mxc91321/pm.c
index feb1f41..f5804fb 100644
--- a/arch/arm/mach-mxc91321/pm.c
+++ b/arch/arm/mach-mxc91321/pm.c
@@ -18,6 +18,9 @@
  * Ported to MXC91321 by Armin Kuster
  * Copyright (c) 2006 MontaVista Software, Inc.
  *
+ * Added .valid hook in mxc_suspend_ops
+ * Copyright 2008 Freescale Semiconductor, Inc. All Rights Reserved.
+ *
  * This program is free software; you can redistribute it and/or modify it
  * under the terms of the GNU General Public License as published by the
  * Free Software Foundation; either version 2 of the License, or (at your
@@ -79,7 +82,13 @@ static void mxc_suspend_finish(void)
 	return;
 }
 
+static int mxc_pm_valid(suspend_state_t state)
+{
+	return (state > PM_SUSPEND_ON && state <= PM_SUSPEND_MAX);
+}
+
 struct platform_suspend_ops mxc_suspend_ops = {
+	.valid = mxc_pm_valid,
 	.prepare = mxc_suspend_prepare,
 	.enter = mxc_suspend_enter,
 	.finish = mxc_suspend_finish,
-- 
1.5.4.4

